DEPLOYING.md

```diff
@@ -0,0 +1,32 @@
+How to Deploy to Maven Central (for Maintainers Only)
+-----------------------------------------------------
+
+If you haven't deployed artifacts to Maven Central before:
+- Follow the instructions on [this
+  page](http://central.sonatype.org/pages/ossrh-guide.html) to set up an
+  account with OSSRH.
+- Install GnuPG and [generate your key pair](https://www.gnupg.org/documentation/howtos.html)
+- Put your GnuPG key password and OSSRH account information in
+  ``<your-home-directory>/.gradle/gradle.properties``:
+```
+signing.keyId=<8-character-public-key-id>
+signing.password=<key-password>
+signing.secretKeyRingFile=<your-home-directory>/.gnupg/secring.gpg
+
+ossrhUsername=<ossrh-username>
+ossrhPassword=<ossrh-password>
+checkstyle.ignoreFailures=false
+```
+
+The Gradle task for deploying artifacts to Maven Central is ``uploadArchives``.
+However, because the codegen, which is a native executable, is built and
+deployed by this task too, you usually need to set additional environment
+variables before running this task. Please refer to _Pushing the codegen to
+Maven Central_ from ``compiler/README.md`` for how to set these environment
+variables on various platforms.
+
+Running ``uploadArchives`` from the top-level project will deploy the codegen
+for one or two platforms, depending on what you put in ``TARGET_ARCHS``. You
+need to run ``uploadArchives`` under ``compiler/`` on the rest of the supported
```

Comment:
The compiler instructions had the command that I could copy & paste ... could we just put that here instead?